File: public/app/monthly-budgets/incoming/payroll/Payroll.ts
export class Payroll {
    private _payPeriod: string;
    private _totalAmount: number;
    private _taxedAmount: number;
    private _netPayment: number;
    private _retirementPlan: number;

    constructor(payrollInfo: any) {
        payrollInfo = JSON.parse(payrollInfo);
        this._payPeriod = payrollInfo.payPeriod;
        this._totalAmount = payrollInfo.totalAmount;
        this._taxedAmount = payrollInfo.taxedAmount;
        this._netPayment = payrollInfo.netPayment;
        this._retirementPlan = payrollInfo.retirementPlan;
    }

    get payPeriod(): string {
        return this._payPeriod;
    }

    get totalAmount(): number {
        return this._totalAmount;
    }

    get taxedAmount(): number {
        return this._taxedAmount;
    }

    get netPayment(): number {
        return this._netPayment;
    }

    get retirementPlan(): number {
        return this._retirementPlan;
    }
}

File: fire-server-core/src/main/java/org/pyhc/fire/payroll/Payroll.java
package org.pyhc.fire.payroll;

import lombok.AccessLevel;
import lombok.Getter;

import java.math.BigDecimal;

@Getter
public class Payroll {

    @Getter(AccessLevel.NONE)
    private String _id;
    private String payPeriod;
    private BigDecimal totalAmount;
    private BigDecimal taxedAmount;
    private BigDecimal netPayment;
    private BigDecimal retirementPlan;

    private Payroll() {
    }

    Payroll(String id, String payPeriod, BigDecimal totalAmount, BigDecimal taxedAmount, BigDecimal netPayment, BigDecimal retirementPlan) {
        this._id = id;
        this.payPeriod = payPeriod;
        this.totalAmount = totalAmount;
        this.taxedAmount = taxedAmount;
        this.netPayment = netPayment;
        this.retirementPlan = retirementPlan;
    }

    public String getId() {
        return _id;
    }
}
